codepropertygraph/src/main/scala/io/shiftleft/utils/StatsCollector.scala

```diff
@@ -0,0 +1,28 @@
+package io.shiftleft.utils
+
+object StatsLogger extends DataLogger {
```

Comment:
Why are we making this an object instead of a class that we can keep all fields immutable?